determined-ai/determined
# fix: change workspace name and set resource quota simultaneously

## Ticket
CM-459

## Description
In the Workspace Edit Modal, we cannot change the workspace name while simultaneously changing the resource quota on the workspace’s auto-created namespace. 

Fix this by storing the workspace’s auto-created namespace in the database when it’s first set, and add a trigger making this column immutable. This means that there will only be once auto-created namespace per workspace going forward, regardless of the workspace’s name changing.


## Test Plan
CI passes (there's automated testing).
We can also manually test this as follows:
In the WebUI:
- create a workspace with an auto-created namespace and resource quota
- Open up the "Edit Workspace" modal for that workspace and change its name and resource-quota simultaneously
- Verify that in Kubernetes, the resource quota in the namespace was changed properly (`kubectl get resourcequotas -n <auto_created_namespace_name>`)


## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.




[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
